apps/client/components/ShopifyPurchaseOrdersTable.tsx

```diff
@@ -0,0 +1,266 @@
+    ComputedPurchaseOrdersRow[]
+  >(
+    purchaseOrders.map((purchaseOrder) => {
+      const { totalCost, quantity } =
+        purchaseOrder.purchase_order_line_items.reduce(
+          (acc, item) => {
+            return {
+              quantity: acc.quantity + item.quantity,
+              totalCost: acc.totalCost + Number(item.unit_cost) * item.quantity,
+            };
+          },
+          { quantity: 0, totalCost: 0 }
+        );
+      return {
+        ...purchaseOrder,
+        id: purchaseOrder.id.toString(),
+        totalCost,
+        quantity,
+      };
+    })
+  );
+
+  const resourceName = {
+    singular: 'order',
+    plural: 'orders',
+  };
+
+  const [sortDirection, setSortDirection] = useState<IndexTableSortDirection>();
+  const [sortedColumnIndex, setSortedColumnIndex] = useState<number>();
+
+  const handleSort = (
+    headingIndex: number,
+    direction: IndexTableSortDirection
+  ) => {
+    setSortDirection(direction);
+    setSortedColumnIndex(headingIndex);
+    const headingKey = [
+      'id',
+      'vendor_name',
+      'order_date',
+      'expected_delivery_date',
+      'quantity',
+      'totalCost',
+    ][headingIndex] as keyof ComputedPurchaseOrdersRow;
+    const sortedRows = [...computedPurchaseOrdersRows].sort((rowA, rowB) => {
+      const valueA = rowA[headingKey];
+      const valueB = rowB[headingKey];
+      if (typeof valueA === 'string') {
+        return valueA.localeCompare(valueB as string);
+      }
+      if (typeof valueA === 'number' && typeof valueB === 'number') {
+        return valueA - valueB;
+      }
+      if (valueA instanceof Date && valueB instanceof Date) {
+        return valueA.getTime() - valueB.getTime();
+      }
+      return 0;
+    });
+    if (direction === 'descending') {
+      sortedRows.reverse();
+    }
+    setComputedPurchaseOrdersRows(sortedRows);
+  };
+
+  const { selectedResources, allResourcesSelected, handleSelectionChange } =
+    useIndexResourceState(computedPurchaseOrdersRows);
+
+  const rowMarkup = useMemo(
+    () =>
+      computedPurchaseOrdersRows.map(
+        (
+          {
+            id,
+            vendor_name,
+            order_date,
+            expected_delivery_date,
+            quantity,
+            totalCost,
+            purchase_order_line_items,
+          },
+          index
+        ) => {
+          console.log(selectedResources);
+          const selected = selectedResources.includes(id.toString());
+          return (
+            <>
+              <IndexTable.Row
+                id={id.toString()}
+                key={id}
+                selected={selected}
+                position={index}
+              >
+                <IndexTable.Cell>
+                  <Text as="span">{id}</Text>
+                </IndexTable.Cell>
+                <IndexTable.Cell>{vendor_name}</IndexTable.Cell>
+                <IndexTable.Cell>
+                  {' '}
```

Comment:
Why is an extra whitespace character being used here?

Reply:
I could honestly remove this, this was a leftover from working out how polaris worked and their Text element didn't play well with potentially null values, could easily address this in a cleaner way in a DateCell component